lib/bgs/vnp_veteran.rb

```diff
@@ -0,0 +1,43 @@
+# frozen_string_literal: true
+
+module BGS
+  class VnpVeteran
+    def initialize(proc_id:, payload:, user:)
+      @user = user
+      @proc_id = proc_id
+      @payload = payload.with_indifferent_access
+      @veteran_info = vnp_veteran.formatted_params(@payload)
+    end
+
+    def create
+      participant = bgs_service.create_participant(@proc_id, nil)
+      claim_type_end_product = bgs_service.find_benefit_claim_type_increment
+      va_file_number = bgs_service.get_va_file_number
+      bgs_service.create_person(person_params(participant))
+      bgs_service.create_phone(@proc_id, participant[:vnp_ptcpnt_id], @veteran_info)
+      address = vet_address(participant[:vnp_ptcpnt_id])
+
+      vnp_veteran.veteran_response(participant, va_file_number, address, claim_type_end_product)
+    end
```

Comment:
How long do you think this will take to finish executing? Seems like it could be a while with all those service calls.

Reply:
@omgitsbillryan Oh yeah, it will take a while.

This is the challenging part of this feature. We have to daisy chain various calls so that we can use ids/values from some to run others.

For example, we are creating a veteran here, We have to use their their `address_id`, `participant_id`, `claim_type_end_product`, and `va_file_number` in subsequent calls where we associate them with their dependents and create a benefit record.

Reply:
All of this will run in a job. PR will be coming in soon

Reply:
Once the job is created, you might be able to delete this class entirely and just put the logic from `def create` into the job.

You'll also have to consider what happens when the job completes - right now at the end of `def create` it calls `vnp_veteran.veteran_response` which looks to be serializing a response. Since jobs don't respond to http requests, you'll probably need to instead save the response into Redis to be fetched in a subsequent GET call to a different endpoint.

Reply:
I'm a bit nervous that `def create` could be a landmine for another developer down the road who doesn't realize it could hang a request for a long time. Once the job is created, you might be able to delete this class entirely and just put the logic from `def create` into the job. That way, there won't be any mistaken uses of this logic.

Reply:
Your concerns are warranted, this is a a wild feature (to me anyway).

I'd like to take a quick step back and make sure we're on the same page, hope you don't mind.

The concept in this PR of creating a 'veteran' is part of many more calls that use the result from this 'veteran'. All of which are run in a single job.

I made a flow chart when we submitted the original big PR that this one derives from. It's a little out dated since we've made changes but I think it still accurately conveys the calls that we do to BGS.

https://user-images.githubusercontent.com/4297274/88064107-d9266280-cb38-11ea-83c2-241b884b2a81.png